### ISSUE TITLE
Fix formatRelative when the unit is set as day

### ISSUE THREAD

**Issue by pith** [formatjs/react-intl#1139]
### Expected behavior
I was expecting the following code to return "tomorrow" when a specify the unit day.
```
intl.formatRelative(
  new Date('2018-06-19T01:00:00'), 
  { units: 'day', now: new Date('2018-06-18T017:00:00') }
)
```
### Current behavior

The above code actually returns "today". 

After few investigations, I understood that the lib is just computing a diff, normally it would have return "8 hours ago". But as the unit is configured with "day", it returns "today" because the diff is less then 24 hours which does make sense in this case.

### Step to reproduce for BUG REPORT

I used the hello world example to reproduce (https://github.com/yahoo/react-intl/tree/master/examples/hello-world)

And in the `App.js`

```
return (
  <p>
    <FormattedRelative 
      value={new Date('2018-06-19T01:00:00)} 
      units='day' 
      initialNow={new Date('2018-06-18T017:00:00')} 
    />
  </p>
);
```

**Comment by jcready** [comment]
Based on your desired output it seems like you would also expect the following to output "next month":

```js
intl.formatRelative(
  new Date('2018-07-01T00:00:00'), 
  { units: 'month', now: new Date('2018-06-30T23:59:59.999') }
)
```

And you would expect this to display "in 2 months" even though it's only 1 month + 1ms away:

```js
intl.formatRelative(
  new Date('2018-08-01T00:00:00'), 
  { units: 'year', now: new Date('2018-06-30T23:59:59.999') }
)
```

**Comment by formatlos** [comment]
I have a similar issue. I'd expect the following code to return "today" when a specify the unit day, but it actually returns tomorrow.

```
intl.formatRelative(
  new Date('2018-06-19T22:00:00'), 
  { units: 'day', now: new Date('2018-06-19T10:00:00') }
)
```

**Comment by matthewgerrard** [comment]
@formatlos I came across the same thing today. There's difference between the dates is calculated `diff.js`. Which rounds the difference between the dates.

If the difference is >=0.5 days it `Math.round`s it to 1 days difference and hence outputs 'tomorrow' - and in the case above is it's 0.5 days.

**Comment by jviall** [comment]
@matthewgerrard I hope they reconsider that rounding behavior. I don't think that anyone would expect displaying the relative time of 7am this morning when it's 9pm at night to show as "Yesterday".

**Comment by longlho** [comment]
fixed in #1301
